dropwizard-logging/src/main/java/io/dropwizard/logging/TlsSocketAppenderFactory.java

```diff
@@ -0,0 +1,429 @@
+ * <td>(none)</td>
+ * <td>
+ * The password used to access the key store.
+ * </td>
+ * </tr>
+ * <tr>
+ * <td>{@code keyStoreType}</td>
+ * <td>{@code JKS}</td>
+ * <td>
+ * The type of key store (usually {@code JKS}, {@code PKCS12}, {@code JCEKS},
+ * {@code Windows-MY}, or {@code Windows-ROOT}).
+ * </td>
+ * </tr>
+ * <tr>
+ * <td>{@code keyStoreProvider}</td>
+ * <td>(none)</td>
+ * <td>
+ * The JCE provider to use to access the key store.
+ * </td>
+ * </tr>
+ * <tr>
+ * <td>{@code trustStorePath}</td>
+ * <td>(none)</td>
+ * <td>
+ * The path to the Java key store which contains the CA certificates used to establish
+ * trust.
+ * </td>
+ * </tr>
+ * <tr>
+ * <td>{@code trustStorePassword}</td>
+ * <td>(none)</td>
+ * <td>The password used to access the trust store.</td>
+ * </tr>
+ * <tr>
+ * <td>{@code trustStoreType}</td>
+ * <td>{@code JKS}</td>
+ * <td>
+ * The type of trust store (usually {@code JKS}, {@code PKCS12}, {@code JCEKS},
+ * {@code Windows-MY}, or {@code Windows-ROOT}).
+ * </td>
+ * </tr>
+ * <tr>
+ * <td>{@code trustStoreProvider}</td>
+ * <td>(none)</td>
+ * <td>
+ * The JCE provider to use to access the trust store.
+ * </td>
+ * </tr>
+ * <tr>
+ * <td>{@code jceProvider}</td>
+ * <td>(none)</td>
+ * <td>The name of the JCE provider to use for cryptographic support.</td>
+ * </tr>
+ * <tr>
+ * <td>{@code validateCerts}</td>
+ * <td>false</td>
+ * <td>
+ * Whether or not to validate TLS certificates before starting. If enabled, Dropwizard
+ * will refuse to start with expired or otherwise invalid certificates.
+ * </td>
+ * </tr>
+ * <tr>
+ * <td>{@code validatePeers}</td>
+ * <td>false</td>
+ * <td>
+ * Whether or not to validate TLS peer certificates.
+ * </td>
+ * </tr>
+ * <tr>
+ * <td>{@code supportedProtocols}</td>
+ * <td>JVM default</td>
+ * <td>
+ * A list of protocols (e.g., {@code SSLv3}, {@code TLSv1}) which are supported. All
+ * other protocols will be refused.
+ * </td>
+ * </tr>
+ * <tr>
+ * <td>{@code excludedProtocols}</td>
+ * <td>[SSL, SSLv2, SSLv2Hello, SSLv3]</td>
+ * <td>
+ * A list of protocols (e.g., {@code SSLv3}, {@code TLSv1}) which are excluded. These
+ * protocols will be refused.
+ * </td>
+ * </tr>
+ * <tr>
+ * <td>{@code supportedCipherSuites}</td>
+ * <td>JVM default</td>
+ * <td>
+ * A list of cipher suites (e.g., {@code TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256}) which
+ * are supported. All other cipher suites will be refused
+ * </td>
+ * </tr>
+ * <tr>
+ * <td>{@code excludedCipherSuites}</td>
+ * <td>[.*_(MD5|SHA|SHA1)$]</td>
+ * <td>
+ * A list of cipher suites (e.g., {@code TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256}) which
+ * are excluded. These cipher suites will be refused.
+ * </td>
+ * </tr>
+ * </table>
+ * <p/>
+ * For more configuration parameters, see {@link TcpSocketAppenderFactory}.
+ *
+ * @see TcpSocketAppenderFactory
+ */
+@JsonTypeName("tls")
+public class TlsSocketAppenderFactory<E extends DeferredProcessingAware> extends TcpSocketAppenderFactory<E> {
```

Comment:
I feel like the TLS configuration code is duplicated in three places: server, client, and now logging. It's fine for this PR, but I'm curious if we could ever consolidate these options so that none of the implementations are ever out of date (like we saw recently with lack of a jce provider for clients)

Reply:
I've been thinking about this too. I guess the biggest issue is that we use different mechanisms for building `SSLContext`'s: Jetty's and Apache HTTP Client's and most of the code is Jackson deserialization code. There's also a mismatch between using it on the server and client sides. I think for the time being, it should be good enough as it is, but we can extract it, if we implement another SSL/TLS client or server.